ebos/eclcpgridvanguard.hh

```diff
@@ -385,6 +384,23 @@ protected:
             }
 
         }
+#if HAVE_MPI
+        {
+            const bool has_numerical_aquifer = this->eclState().aquifer().hasNumericalAquifer();
+            int mpiSize = 1;
+            MPI_Comm_size(MPI_COMM_WORLD, &mpiSize);
+            // when there is numerical aquifers, new NNC are generated during grid processing
+            // we need to pass the NNC from root process to other processes
+            if (has_numerical_aquifer && mpiSize > 1) {
+                auto nnc_input = this->eclState().getInputNNC();
+                Opm::EclMpiSerializer ser(Dune::MPIHelper::getCollectiveCommunication());
+                ser.template broadcast(nnc_input);
```

Comment:
Do you really need `.template` here?  For one, there's no explicit template argument.  Secondly, I don't really see how `nnc_input` depends on a template parameter.

Reply:
Yes. It looks like `template` is not needed. But I do not recall why it is there in the first place either. I will update it.